### ISSUE TITLE
The future of unit control should be production based.

### ISSUE THREAD

**Issue by Larofeticus** [Anuken/Mindustry-Suggestions#3427]
### Describe the content or mechanics you are proposing.

Issuing RTS style commands to units should require and be limited by infrastructure, resources, and production.

An example from elsewhere in gaming is the "Command Point" system in Starsector. It's a space fleet RPG with tactical fleet battles; the player can directly control one ship, but giving orders beyond the default behavior to other ships in fleet consumes a limited number of "Command Points" which regenerate over time.

So, first addition would be a resource that is consumed when giving commands. Some plausible names would be "bandwidth" "command" or "control". Thematically, units could be said to have authentication/firewalls to ensure only orders from the owner player are accepted, this resource could be something from cybersecurity or cryptography vocabulary like "hashes" "tokens" or "keys"

The command resource is generated in factories ("computer" "mainframe" something like that) that has power and material requirements. There are also logistics to transport and store command, things from the vocabulary of information technology.

Command resource is loaded into a building which consumes it when the player gives orders to units. Something like "antenna" "communications dish" "broadcast tower". These buildings have a radius, and a rate at which they can consume command to give orders; more antenna would be needed to cover more of the map, or give orders to more units per unit of time.

The player units would have some small amount of intrinsic command range and ability, enough to move a couple daggers on purpose, but not enough to attack-move an entire army.

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

I believe this is a very mindustry thematic way to give some intuitive player control of units without making it fully an RTS game of which better examples already exist.

Building economy, logistics, infrastructure, and production to command units is playing to mindustry identity in a way that logic unit control was not.

If unit control was not somehow limited by rate or buildings then the game would devolve into being entirely unit vs unit. Why bother building and feeding tons of surge into a foreshadow when you can get unlimited tactical corvus behavior for free?

The command resource could be more widely used if you like; unit or building boosts/efficiency. Unit buffs. Turrets that use command as ammo and hack enemy units to debuff, disable, or even steal them.

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by Anuken** [comment]
> If unit control was not somehow limited by rate or buildings then the game would devolve into being entirely unit vs unit. 

First of all, why is this necessarily bad? It's significantly more interesting to move units around a map to repel attacks than it is to simply place down a turret and wait for something to happen. Turrets do have to be rebalanced to account for smarter units, but RTS games *do* have turrets, and they work fine for their intended purpose.

> when you can get unlimited tactical corvus for free

It really sounds like the problem here is *unlimited ammo*, not behavior. If units required ammo (which *is* a game rule), you wouldn't get anything "for free" - units would require production to work properly, just like turrets - except without entirely new mechanics being introduced into the game.

**Comment by Larofeticus** [comment]
Well, I'll repeat this argument first since it's one of the stronger ones: If a player primarily wants an actions-per-minute micro unit vs unit RTS game, then they'll be playing starcraft or some other genre tentpole and not Mindustry. 

It would be bad for Mindustry gameplay to become "entirely unit vs unit" because most of the content in your game is buildings and building. The campaign primarily teaches players to use buildings. RTS games will have 1 or 2 static defense buildings per faction, while you've created 18. You don't yet have differentiated enough units or damage rock-paper-scissors for multiple viable army compositions, or factions, to support unit vs unit being primary, and it would take a lot of effort to get there.

The role of turrets is to put defense in place that does not require player attention to function. Sometimes I don't want to be actively managing an army, I want to be buying more safety to build economy or do something else; those choices should to continue to be viable. I'm fine with moving units to augment defense (minus the free ammo issue, you already know it) but I don't want an outcome where intense micro on units can allow them to punch way above weight (as that being _possible_ would make it _necessary_ to be competitive)

Would the prototype unit selection in your demo also be able to select and control multiple turrets at the same time?

I'd also support finishing the unit ammo system.

**Comment by Anuken** [comment]
> If a player primarily wants an actions-per-minute micro unit vs unit RTS game, then they'll be playing starcraft or some other genre tentpole and not Mindustry.

Mindustry's production and logistics aspect makes it very different from a standard RTS game, so I don't follow.

> RTS games will have 1 or 2 static defense buildings per faction

Planetary Annihilation - one of the more modern RTS games I have played - has 15. Warzone 2100, another (old) RTS I played a while back has far more. Neither games have a concept of factions outside of mods, so I'm not sure where you're getting this from.

> You don't yet have differentiated enough units or damage rock-paper-scissors for multiple viable army compositions, or factions, to support unit vs unit being primary

I don't see why this isn't at least partially achievable.

> I don't want an outcome where intense micro on units can allow them to punch way above weight 

In my experience, that doesn't happen at all.

> Would the prototype unit selection in your demo also be able to select and control multiple turrets at the same time?

No, there is little need for that, as turrets don't move and usually choose the best unit target anyway. The only times I've actually wanted to control a turret directly is when it was targeting enemy buildings... and this is, in my opinion, a degenerate tactic that should be eliminated, not encouraged.

> You made a tower defense game; 

Did I, though? Mindustry's "tower defense" meta is to wall off the enemy spawn and kill them immediately. It's a flat DPS/regen/production speed check; enemies do not get the chance to move along a path. This is not how games of the genre tend to work.

> Campaign experience and difficulty pivots on that;

This hinges on the assumption that the Erekir campaign will be based upon tower defense... which it doesn't need to be.

> You won't be able to write attack/unit AI equivalent to the skill of a human opponent

Not exactly, no, but there's plenty that can be done to make it *good enough* to pose a challenge. Many RTS games seem to manage this just fine.

> then those same turrets will be massively overpowered against automated campaign content.

First of all, "automated campaign content" implies that I will be using the brain-dead tower defense AI - which, again, I won't be.

Second of all - Why? Even if the campaign *was* entirely tower defense, I control *everything the enemy sends at you*. It doesn't matter how overpowered the turrets are when I can just compensate by using more powerful or more numerous units. This is not an issue.

AI cheating is *very* common in RTS games. There's absolutely no need for the player and campaign AI to start on even footing or play by the exact same rules.
